src/Features/Core/Portable/Diagnostics/WorkspaceAnalyzerOptions.cs

```diff
@@ -15,13 +15,11 @@ namespace Microsoft.CodeAnalysis.Diagnostics
     internal sealed class WorkspaceAnalyzerOptions : AnalyzerOptions
```

Comment:
>WorkspaceAnalyzerOptions [](start = 26, length = 24)

Is this class still needed?

Reply:
Our IDE analyzers still need the options that are stored inside here, and also use the [Document.GetOptionsAsync](http://sourceroslyn.io/#Microsoft.CodeAnalysis.Features/Diagnostics/WorkspaceAnalyzerOptions.cs,44).

Reply:
This type is very odd. Seems like a sneaky way to smuggle Workspace and Solution to the analyzers. I think this will need to go at some point but we can do that later.

---
In reply to: [358490994](https://github.com/dotnet/roslyn/pull/40289#discussion_r358490994) [](ancestors = 358490994)

Reply:
Yes, it should eventually go away: https://github.com/dotnet/roslyn/issues/40463